### PR TITLE
Meistejw/feature/014/backend portion of add positive case

### LINKED ISSUE #10: Connect the front end and the backend

### LINKED ISSUE #14: Create a Positive Case
As a member of health services, I want to be able to indicate a positive case by selecting a student by name or ID who has tested positive. 

**Acceptance Criteria**
- [ ] Member of health services can select a student by ID or name 
- [ ] Member of health services can update the student's information to include that they have a positive case 
- [ ] Student is added to the list of positive cases

### LINKED ISSUE #20: Create a Positive case table

### LINKED ISSUE #25: Implement Spring Annotations for Spring Data
As a developer, I want to use Spring's persistence annotations to format classes into Spring Data's format in order to make the system work with Java Persistence API and Controller/Service/Repository pattern. 

**Acceptance Criteria: **

- [ ]  Model classes have Spring Data annotations (eg. Entity, Table, Column, etc.) 
- [ ]  Repository classes extend Spring Data Repositories (eg. PagingAndSortingRepository) 
- [ ] System uses @Autowiring to inject Services, Repositories, Controllers, etc where needed

### PR DESCRIPTION
**Pull request by meistejw**
Got a little carried away and added a few more api endpoints than just add a case. This branch essentially covers changes that were needed as I was connecting the frontend to the backend. There were also some left over issues that need to be closed that relate to the Spring annotations in each level of our application and I definitely made some changes to those in this branch.

closes #10 
closes #14 
closes #20 
closes #25

### CODE REVIEW

File: src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java [hunk 1]
```diff
@@ -1,7 +1,108 @@
...
+	public PositiveCase retrievePositiveCase(@PathVariable Long id) throws NotFoundException {
+		Optional<PositiveCase> positiveCase = repository.findById(id);
+
+		if (!positiveCase.isPresent())
+			throw new NotFoundException("id-" + id);
+
+		return positiveCase.get();
+	}
+
+	@PostMapping("/userlist/{id}")
```
Comment (noahthomas88) [inline src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java:63]: So the Get mapping seems to be the get API endpoint and the post is the post endpoint. Very clean.

File: src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java [hunk 2]
```diff
@@ -1,7 +1,108 @@
...
+			@RequestBody PositiveCase positiveCase) throws NotFoundException {
+		return personRepo.findById(personID).map(person -> {
+			positiveCase.setPerson(person);
+			return repository.save(positiveCase);
+		}).orElseThrow(() -> new NotFoundException("PersonID" + personID + " not found"));
+
+	}
+	
+	@CrossOrigin(origins = "http://localhost:4200", maxAge = 3600)
+	@PutMapping("/positive-case/{id}")
```
Comment (noahthomas88) [inline src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java:89]: This is the endpoint to add a contact tracer?

File: src/main/java/edu/rosehulman/covidtracer/model/PositiveCase.java [hunk 3]
```diff
@@ -40,11 +41,12 @@
 	public PositiveCase() {
 	}
 
-	public PositiveCase(Person person, Date testDate, Date symptomaticStartDate, Date quarantineEndDate, boolean needsTransportation) {
+	public PositiveCase(Long ID, Person person, Date d, Date d2, Date d3, boolean needsTransportation) {
```
Comment (noahthomas88) [inline src/main/java/edu/rosehulman/covidtracer/model/PositiveCase.java]: Should the names for the dates in the constructor be more descriptive?

File: src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java [hunk 4]
```diff
@@ -33,6 +34,8 @@
 
 	@Column(name = "isClean", nullable = false)
 	private boolean isClean = true;
+	
+	public QuarantineLocation() {}
```
Comment (noahthomas88) [inline src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java:38]: What does this code do/where is it used?

File: src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java (hunk 2 above)
Comment (meistejw) [reply src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java:89]: Yes.

File: src/main/java/edu/rosehulman/covidtracer/model/PositiveCase.java (hunk 3 above)
Comment (meistejw) [reply src/main/java/edu/rosehulman/covidtracer/model/PositiveCase.java]: Oh, oops. That constructor was auto generated and I never changed it. Good catch!

File: src/main/java/edu/rosehulman/covidtracer/service/PositiveCaseService.java [hunk 5]
```diff
@@ -21,13 +25,18 @@
 	@Autowired
     PositiveCaseRepository repository;
 
-    public List<PositiveCase> getAllStudents(){
-        Pageable pageProperties = PageRequest.of(0, 10, Sort.by("id"));
+    public List<PositiveCase> getAllPositiveCases(Integer pageNum, Integer entriesPerPage, String sortBy){
```
Comment (noahthomas88) [inline src/main/java/edu/rosehulman/covidtracer/service/PositiveCaseService.java]: Are pageNum or entriesPerPage used in this?

File: src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java (hunk 4 above)
Comment (meistejw) [reply src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java:38]: It is needed by Spring in order to generate the objects at startup. This is how spring verifies/builds these entities in the database.

**Review by noahthomas88 — CHANGES_REQUESTED** [review]
See the comments on descriptive dates in a constructor and using all the params in the getAllPositiveCases().

File: src/main/java/edu/rosehulman/covidtracer/service/PositiveCaseService.java (hunk 5 above)
Comment (meistejw) [reply src/main/java/edu/rosehulman/covidtracer/service/PositiveCaseService.java]: Hmmm. These are the values that make our GETS limited to 10 items per page. The values are received from the frontend and they determine how much data is sent at once. We might need to discuss as a team whether or not we want to use pages or not.

File: src/main/java/edu/rosehulman/covidtracer/model/PositiveCase.java (hunk 3 above)
Comment (noahthomas88) [reply src/main/java/edu/rosehulman/covidtracer/model/PositiveCase.java]: Thanks m8! :D

File: src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java (hunk 4 above)
Comment (meistejw) [reply src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java:38]: It is needed by Spring in order to generate the objects at startup. This is how spring verifies/builds these entities in the database.

File: src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java (hunk 2 above)
Comment (meistejw) [reply src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java:91]: Yes.

File: src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java (hunk 4 above)
Comment (noahthomas88) [reply src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java:38]: But are we using this particular constructor anywhere in the code?

Comment (noahthomas88) [reply src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java:38]: Or was it autogenerated and do we need to remove it?

Comment (meistejw) [reply src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java:38]: For whatever reason, Spring gets mad if you don't have it. Don't the other classes have blank constructors in them?

File: src/main/java/edu/rosehulman/covidtracer/service/PositiveCaseService.java (hunk 5 above)
Comment (noahthomas88) [reply src/main/java/edu/rosehulman/covidtracer/service/PositiveCaseService.java]: I think pages are fine to go with for now. I think the users should decide weather they like that user experience.

File: src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java (hunk 4 above)
Comment (noahthomas88) [reply src/main/java/edu/rosehulman/covidtracer/model/QuarantineLocation.java:38]: Yeah they do, I apologize, didn't notice that was a feature of spring

**Review by noahthomas88 — APPROVED** [review]
new changes Look good! Lets merge this! :D

**Pull request merged**